apis/HdmiCecSource/IHdmiCecSource.h

```diff
@@ -0,0 +1,172 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2024 Synamedia Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "Module.h"
+
+// @stubgen:include <com/IIteratorType.h>
+
+namespace WPEFramework
+{
+    namespace Exchange
+    {
+        // @json @text:keep
+        struct EXTERNAL IHdmiCecSource : virtual public Core::IUnknown {
+            enum { ID = ID_HDMI_CEC_SOURCE };
+
+            struct HdmiCecSourceDevices
+            {
+                uint8_t logicalAddress;
+                string vendorID;
+                string osdName;
+            };
+
+            using IHdmiCecSourceDeviceListIterator = RPC::IIteratorType<HdmiCecSourceDevices, ID_HDMI_CEC_SOURCE_DEVICE_LIST_ITERATOR>;
+
+
+            struct EXTERNAL INotification : virtual public Core::IUnknown
+            {
+                enum { ID = ID_HDMI_CEC_SOURCE_NOTIFICATION };
+    
+                // @brief Device added event
+                // @text onDeviceAdded
+                // @param logicalAddress: Logical address of the added device
+                virtual void OnDeviceAdded(const int logicalAddress) {};
+    
+                // @brief Device removed event
+                // @text onDeviceRemoved
+                // @param logicalAddress: Logical address of the removed device
+                virtual void OnDeviceRemoved(const int logicalAddress /* @in */) {};
+    
+                // @brief Device info updated event
+                // @text onDeviceInfoUpdated
+                // @param deviceInfo: Device info of the updated device
+                virtual void OnDeviceInfoUpdated(const int logicalAddress/* @in */) {};
+    
+                // @brief Active source status updated event
+                // @text onActiveSourceStatusUpdated
+                // @param isActiveSource: Is the active source active or not
+                virtual void OnActiveSourceStatusUpdated(const bool status /* @in */) {};
+    
+                // @brief Triggered when a device enters standby
```

Comment:
Terred when CEC standby message received from the other CEC device

Reply:
Updated